zerver/lib/actions.py

```diff
@@ -4477,17 +4477,25 @@ def check_change_bot_full_name(
 def do_change_bot_owner(
     user_profile: UserProfile, bot_owner: UserProfile, acting_user: UserProfile
 ) -> None:
+    if user_profile.can_create_users and not acting_user.is_realm_owner:
+        raise JsonableError(
+            _(
+                "Only organization owners can change ownership of a bot with the can_create_users permission."
+            )
+        )
```

Comment:
I think we want this check in the ACL code for modifying this bot at all.  Generally we never adds ACL code in `do_` functions, as a naming convention.  I think this function is where we want it:
```
def access_bot_by_id(user_profile: UserProfile, user_id: int) -> UserProfile:                            
    try:                                                                                                 
        target = get_user_profile_by_id_in_realm(user_id, user_profile.realm)                            
    except UserProfile.DoesNotExist:                                                                     
        raise JsonableError(_("No such bot"))                                                            
    if not target.is_bot:                                                                                
        raise JsonableError(_("No such bot"))                                                            
    if not user_profile.can_admin_user(target):                                                          
        raise JsonableError(_("Insufficient permission"))                                                
    return target                                                                                        

```

Reply:
I think it's logically simplest to just have "you can administer this user or not". Using common functions like this is really important from an organization architecture standpoint, because then we don't need to think about security bugs along the lines of some new feature that we think requires "ability to administer this bot" escaping this restriction.

Also switched the exception to just  `raise OrganizationOwnerRequired()`